WriteVHDLSyntax.py

```diff
@@ -634,8 +631,54 @@ def writeTopLevelMemoryType(mtypeB, memList, memInfo, extraports, delay = 0, spl
             delay_portlist += "        wea_out       => "+mem+"_wea_delay,\n"
             delay_portlist += "        addra_out     => "+mem+"_writeaddr_delay,\n"
             delay_portlist += "        dina_out      => "+mem+"_din_delay,\n"
-        
-
+        # add merge_stream modules if split and TPAR or AS
+        if "TPAR" in mem and split == 1:
+            addrwidth = 10
+            ramwidth = memInfo.bitwidth
+            numpages = 8
+            seed = mem.split("_")[1][:-1]
+            iTC = mem.split("_")[1][-1]
+            for PCGroup in MPARdict[seed]:
+                if iTC == PCGroup[0]:
+                    numInputs = len(PCGroup)
+                    merge_parameterlist += "        RAM_WIDTH => "+str(ramwidth)+",\n"
+                    merge_parameterlist += "        NUM_PAGES => "+str(numpages)+",\n"
+                    merge_parameterlist += "        NUM_INPUTS => "+str(numInputs)+",\n"
+                    merge_parameterlist += "        NUM_EXTRA_BITS => 2,\n"
+                    merge_portlist += "        bx_in => TP_bx_out,\n"
+                    merge_portlist += "        rst => '0',\n"
+                    merge_portlist += "        clk => clk,\n"
+                    merge_portlist += "        enb_arr => open,\n"
+                    merge_portlist += "        bx_out => open,\n"
+                    merge_portlist += "        merged_dout => MPAR_"+seed+PCGroup+"_stream_V_dout,\n"
+                    for i in range(4):  merge_portlist += "        din"+str(i)+"=>TPAR_"+seed+PCGroup[i%numInputs]+"_V_dout,\n"
+                    for i in range(4):  merge_portlist += "        nent"+str(i)+"=>TPAR_"+seed+PCGroup[i%numInputs]+"_AV_dout_nent,\n"
+                    for i in range(numInputs):  merge_portlist += "        addr_arr("+str(((i+1)*addrwidth)-1)+" downto "+ str(i*addrwidth)+ ")=>TPAR_"+seed+PCGroup[i%numInputs]+"_V_readaddr,\n"
```

Comment:
@jasonfan393 Would it be possible to eliminate the hard-coded value of `addrwidth`, e.g., by concatenating the inputs to `addr_arr`?:
```suggestion
            ramwidth = memInfo.bitwidth
            numpages = 8
            seed = mem.split("_")[1][:-1]
            iTC = mem.split("_")[1][-1]
            for PCGroup in MPARdict[seed]:
                if iTC == PCGroup[0]:
                    numInputs = len(PCGroup)
                    merge_parameterlist += "        RAM_WIDTH => "+str(ramwidth)+",\n"
                    merge_parameterlist += "        NUM_PAGES => "+str(numpages)+",\n"
                    merge_parameterlist += "        NUM_INPUTS => "+str(numInputs)+",\n"
                    merge_parameterlist += "        NUM_EXTRA_BITS => 2,\n"
                    merge_portlist += "        bx_in => TP_bx_out,\n"
                    merge_portlist += "        rst => '0',\n"
                    merge_portlist += "        clk => clk,\n"
                    merge_portlist += "        enb_arr => open,\n"
                    merge_portlist += "        bx_out => open,\n"
                    merge_portlist += "        merged_dout => MPAR_"+seed+PCGroup+"_stream_V_dout,\n"
                    for i in range(4):  merge_portlist += "        din"+str(i)+"=>TPAR_"+seed+PCGroup[i%numInputs]+"_V_dout,\n"
                    for i in range(4):  merge_portlist += "        nent"+str(i)+"=>TPAR_"+seed+PCGroup[i%numInputs]+"_AV_dout_nent,\n"
                    merge_portlist += "        addr_arr=>(TPAR_"+seed+PCGroup[-1]+"_V_readaddr"
                    for i in reversed(range(numInputs-1)):  merge_portlist += ", TPAR_"+seed+PCGroup[i]+"_V_readaddr"
                    merge_portlist += "),\n"
```